ksql-common/src/main/java/io/confluent/ksql/config/KsqlConfigResolver.java

```diff
@@ -77,7 +77,8 @@
 
     if (propertyName.startsWith(KSQL_STREAMS_PREFIX)
         && !propertyName.startsWith(KSQL_STREAMS_PREFIX + StreamsConfig.PRODUCER_PREFIX)
-        && !propertyName.startsWith(KSQL_STREAMS_PREFIX + StreamsConfig.CONSUMER_PREFIX)) {
+        && !propertyName.startsWith(KSQL_STREAMS_PREFIX + StreamsConfig.CONSUMER_PREFIX)
+        && !propertyName.startsWith(KSQL_STREAMS_PREFIX + StreamsConfig.TOPIC_PREFIX)) {
       return Optional.empty();  // Unknown streams config
```

Comment:
out of scope for this PR, but I'm wondering why we don't let people pass in any arbitrary streams config and let streams do the verification here

Reply:
we seem to have invested a lot into resolving configs in specific ways. I don't know the full context. But yeah, we could keep it simpler and pass everything with prefix to  streams. But as of now, there are only these three configs that Streams documents. So may be ok for now.